### PR TITLE
Feat/ Adjust SKS error handling && add REST API caching

### PR DESCRIPTION
**Pull request by simon-the-shark**
1. Adjusted error handling
  - previously we had shown sks animation with any error
  - now we show:
      - sks animation when the menu is not available (that's not really an exception, just a normal situation)
      - offline animation when we don't have internet connection
      - our `MyErrorWidget` on any other error

image
image

2. Added caching
 - we cache the data for 1 day, only when the available menu is up to date
 - we always invalidate the cache when the cached data is from the previous day (after midnight)

3. pull to refresh
    - we invalidate the cache and redo the request

Zrzut ekranu 2024-12-10 o 13 31 06

### CODE REVIEW

**Review by tomasz-trela — COMMENTED** [review]
Good job. I wrote down one suggestion.

File: lib/features/sks-menu/data/repository/sks_menu_repository.dart [hunk 1]
```diff
@@ -1,36 +1,51 @@
...
+    );
+
+    final trueMeals = sksMenuResponse.meals
+        .where((e) => e.category != DishCategory.technicalInfo)
+        .toList();
+
+    final technicalInfos = sksMenuResponse.meals
+        .where((e) => e.category == DishCategory.technicalInfo)
+        .map((e) => e.name)
+        .toList();
```
Comment (tomasz-trela) [inline lib/features/sks-menu/data/repository/sks_menu_repository.dart:42]: Consider changing to IList.

**Review by tomasz-trela — COMMENTED** [review]
^

File: lib/features/sks-menu/data/repository/sks_menu_repository.dart (hunk 1 above)
Comment (simon-the-shark) [reply lib/features/sks-menu/data/repository/sks_menu_repository.dart:42]: if only freezed supports this then I'm for it

Comment (simon-the-shark) [reply lib/features/sks-menu/data/repository/sks_menu_repository.dart]: done, thanks

**Pull request merged**